@here/harp-mapview/lib/text/Placement.ts

```diff
@@ -430,62 +422,63 @@ function placePointLabelChoosingAnchor(
-    // Provide private copy of label layout and attach it to current label and text canvas.
-    label.layoutStyle = label.layoutStyle!.clone();
-    textCanvas.textLayoutStyle = label.layoutStyle!;
-
     // The current implementation does not provide placements options via theme style yet,
     // so function tries the anchor placements from pre-defined placements arrays.
     // TODO: HARP-6487 Placements options should be loaded from the theme.
 
-    // Start with currently set alignment settings.
-    const basePlacement = {
-        h: label.layoutStyle!.horizontalAlignment,
-        v: label.layoutStyle!.verticalAlignment
-    };
+    // Start with recently set alignment settings if layout state is stored or
+    // simply begin from layout defined in theme.
+    const labelThemeLayout = label.layoutStyle!;
+    const labelStateLayout = labelState.textLayoutState;
+    const recentPlacement =
+        persistent && labelStateLayout !== undefined
+            ? labelStateLayout.getPlacement(labelThemeLayout)
+            : { h: labelThemeLayout.horizontalAlignment, v: labelThemeLayout.verticalAlignment };
     const placementCentered =
-        basePlacement.h === HorizontalAlignment.Center ||
-        basePlacement.v === VerticalAlignment.Center;
-    // Placements options will be read from label.layoutStyle.placements in final solution.
+        recentPlacement.h === HorizontalAlignment.Center ||
+        recentPlacement.v === VerticalAlignment.Center;
+    // TODO: HARP-6487: Read placements options from label.layoutStyle.placements.
     const placements = placementCentered ? anchorPlacementsCentered : anchorPlacementsCornered;
     const placementsNum = placements.length;
     // Find current anchor placement on the optional placements list.
     // Index of exact match.
-    const matchIdx = placements.findIndex(p => p.h === basePlacement.h && p.v === basePlacement.v);
+    const matchIdx = placements.findIndex(
+        p => p.h === recentPlacement.h && p.v === recentPlacement.v
+    );
     assert(matchIdx >= 0);
     // Will be true if all text placements are invisible.
     let allInvisible: boolean = true;
     // Iterate all placements starting from current one.
     for (let i = matchIdx, tryIdx = 0; tryIdx < placementsNum; ++i, ++tryIdx) {
         const anchorPlacement = placements[i % placementsNum];
-        // Override label text alignment for measurements and leave it so if passed collisions test.
-        // NOTE: This actually writes also to:
-        // textCanvas.textLayoutStyle.verticalAlignment/horizontalAlignment.
-        // which is needed for text measurements being done on canvas.
-        label.layoutStyle!.horizontalAlignment = anchorPlacement.h;
-        label.layoutStyle!.verticalAlignment = anchorPlacement.v;
-
-        // Bounds may be already calculated for persistent label, force re-calculation for
-        // for alternative placements.
-        const basePlacementTest = tryIdx === 0;
+
```

Comment:
tryIdx is completely unneeded. tryIdx === i - matchIdx. It's just making the loop more complicated to understand than it should be.

Reply:
Ok I will do so, but I am not sure if it's gonna be simpler to read... Keep in mind that we need to iterate as in round buffer.